app/assets/javascripts/short-form/components/lendingInstitution.js.coffee

```diff
@@ -0,0 +1,46 @@
+angular.module('dahlia.components')
+.component 'lendingInstitution',
+  bindings:
+    application: '<'
+    form: '<'
+  templateUrl: 'short-form/components/lending-institution.html'
+  controller:
+    ['ShortFormApplicationService', 'ShortFormHelperService', '$translate',
+    (ShortFormApplicationService, ShortFormHelperService, $translate) ->
+      ctrl = @
+      @lendingInstitutions = ShortFormApplicationService.lendingInstitutions
+      @selectedInstitution = ''
+      @officers = []
+
+      @$onInit = ->
+        if @application.lendingAgent
+          @selectedInstitution = _.findKey(@lendingInstitutions, (agents) ->
+            _.some(agents, {'Id': ctrl.application.lendingAgent})
+          )
+          @officers = @lendingInstitutions[@selectedInstitution]
+
+      @inputInvalid = (fieldName) ->
+        form = @form.applicationForm
+        return false unless form
+        field = form[fieldName]
+        if form && field
+          field.$invalid && (field.$touched || form.$submitted)
+        else
+          false
+
+      @lendingInstitutionsNames = ->
+        _.keys(@lendingInstitutions)
+
+      @onChangeLendingInstitution = ->
+        @application.lendingAgent = null
+        @officers = @lendingInstitutions[@selectedInstitution]
+
+      @officesSelectActive = ->
```

Comment:
looks like a typo in the name of this function. I'm not super clear on what its purpose is either